ahnlich/server/src/main.rs

```diff
@@ -5,10 +5,85 @@ mod engine;
 mod errors;
 mod network;
 mod storage;
+use std::error::Error;
 
-#[cfg(test)]
-mod tests;
+use clap::{ArgAction, Args, Parser, Subcommand};
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::net::{TcpListener, TcpStream};
+
+async fn run_server() -> std::io::Result<()> {
+    let listener = TcpListener::bind("127.0.0.1:1369").await?;
 
-fn main() {
-    println!("Hello, world!");
+    loop {
+        let (stream, connect_addr) = listener.accept().await?;
+        println!("Connecting to {}", connect_addr);
```

Comment:
Most likely introduce log in the workspace instead of using any print or println so the log level can be easily toggled